tests/webTables.spec.ts

```diff
@@ -0,0 +1,202 @@
+    ).toHaveText(petName);
+  });
+
+  test("Validate pets of the Madison city", async ({ page }) => {
+    const madisonPetRows = page.locator("tbody tr", { hasText: "Madison" });
+    const pets = [];
+
+    await expect(
+      page.locator("tbody tr").filter({ hasText: "Madison" })
+    ).toHaveCount(4);
+
+    for (let row of await madisonPetRows.all()) {
+      pets.push(
+        (await row.locator("td").last().textContent()).toString().trim()
+      );
+    }
+    expect(pets).toEqual(["Leo", "George", "Mulligan", "Freddy"]);
+  });
+});
+
+test.describe("Web Tables-Veterinarians", async () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto("/");
+    await page.getByText("Veterinarians").click();
+    await page.getByText("All").click();
+  });
+
+  test("Validate specialty update", async ({ page }) => {
+    const targetRow = page.getByRole("row", { name: "Rafael Ortega" });
+    await expect(targetRow.locator("td").nth(1)).toHaveText("surgery");
+
+    await page.getByRole("link", { name: "Specialties" }).click();
+
+    await expect(page.locator("h2")).toHaveText("Specialties");
+
+    await page
+      .getByRole("row", { name: "surgery" })
+      .getByRole("button", { name: "Edit" })
+      .click();
+
+    await expect(page.locator("h2")).toHaveText("Edit Specialty");
+
+    const specialtyNameInputField = page.getByRole("textbox");
+    await specialtyNameInputField.click();
+    await specialtyNameInputField.clear();
+    await specialtyNameInputField.fill("dermatology");
+    await page.getByRole("button", { name: "Update" }).click();
+
+    expect(page.getByRole("row", { name: "dermatology" }));
+
+    await page.getByText("Veterinarians").click();
+    await page.getByText("All").click();
+    await expect(targetRow.locator("td").nth(1)).toHaveText("dermatology");
+
+    await page.getByRole("link", { name: "Specialties" }).click();
+    await page
+      .getByRole("row", { name: "dermatology" })
+      .getByRole("button", { name: "Edit" })
+      .click();
+    await specialtyNameInputField.click();
+    await specialtyNameInputField.clear();
+    await specialtyNameInputField.fill("surgery");
+    await page.getByRole("button", { name: "Update" }).click();
+  });
+});
+
+test.describe("Web Tables-Specialties", async () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto("/");
+    await page.getByText("Specialties").click();
+  });
+
+  test("Validate specialty lists", async ({ page }) => {
+    await page.getByRole("button", { name: "Add" }).click();
+    await page.locator("#name").fill("oncology");
+    await page.getByRole("button", { name: "Save" }).click();
+
+    await page.waitForTimeout(1000);
+
+    const specialties = [];
+    const specialtyRows = page.locator("tbody tr input");
+
+    for (let specialty of await specialtyRows.all()) {
+      const specialtyName = await specialty.inputValue();
+      specialties.push(specialtyName);
```

Comment:
You can use the exclamation sign "!" for the variable to tell TypeScript that it's not null

Reply:
done